### PR TITLE
Async xpath

### PR DESCRIPTION
**Pull request by DrRataplan**

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -52,6 +51,8 @@ function doSelectorsBuild () {
 			warning_level: 'VERBOSE',
 			compilation_level: 'ADVANCED',
 			externs: path.resolve('./externs/IDomFacade.js'),
+			module_resolution: 'NODE',
```
Comment (DrRataplan) [inline build.js]: Note: we'll change this back to browser after a replace all of `s/import ... from ...';/import ... from ...**.js**';

File: test/unrunnableTestCases.csv [hunk 2]
```diff
@@ -1928,17 +1869,17 @@ rangeExpr-37,Error: XPST0017: Function year-from-date with arity of 1 not regist
...
+K-RangeExpr-31,Error: XPST0003: Unable to parse XPath: ((remove((2.e0, 4), 1) treat as xs:integer to 4))!string() => string-join(" "). SyntaxError: Expected " ", "!", "!=", "(", "(:", ")", "*", "+", ",", "-", "/", "//", "<", "<<", "<=", "=", "=>", ">", ">=", ">>", "?", "[", "\n", "\r", "\t", "and", "cast", "castable", "div", "eq", "except", "ge", "gt", "idiv", "instance", "intersect", "is", "le", "lt", "mod", "ne", "or", "to", "union", "|", or "||" but "t" found.
+K-RangeExpr-32,Error: XPST0003: Unable to parse XPath: ((4 to remove((2e0, 4), 1) treat as xs:integer))!string() => string-join(" "). SyntaxError: Expected " ", "!", "!=", "(", "(:", ")", "*", "+", ",", "-", "/", "//", "<", "<<", "<=", "=", "=>", ">", ">=", ">>", "?", "[", "\n", "\r", "\t", "and", "cast", "castable", "div", "eq", "except", "ge", "gt", "idiv", "instance", "intersect", "is", "le", "lt", "mod", "ne", "or", "union", "|", or "||" but "t" found.
 fn-union-node-args-003,Error: XPST0017: Function root with arity of 1 not registered. No similar functions found.
+fn-union-node-args-006,TypeError: Cannot read property '0' of undefined
```
Comment (DrRataplan) [inline test/unrunnableTestCases.csv]: This is a regression

File: test/unrunnableTestCases.csv [hunk 3]
```diff
@@ -2320,6 +2246,7 @@ instanceof131,Error: No selector counterpart for: functionTest.
 instanceof132,Error: No selector counterpart for: functionTest.
 instanceof133,Error: No selector counterpart for: functionTest.
 instanceof134,Error: No selector counterpart for: functionTest.
+K-SeqExprInstanceOf-18,AssertionError: Expected XPath not((1, 2, 3, 4, 5) instance of xs:integer) to resolve to true: expected false to be true
```
Comment (DrRataplan) [inline test/unrunnableTestCases.csv]: This is a regression

Comment (bwrrp) [inline test/unrunnableTestCases.csv]: regression?

**Pull request merged**